File: sdks/typescript/openfoundry-sdk/src/examples/phase1-critical-path.ts
import { OpenFoundryClient, type ControlPanelSettings } from '../index';

export async function phase1CriticalPath(client: OpenFoundryClient): Promise<{
	datasets: number;
	ontologyTypes: number;
	pipelines: number;
	controlPanel: ControlPanelSettings;
}> {
	const [datasets, ontology, pipelines, controlPanel] = await Promise.all([
		client.datasetDatasetListdatasets(),
		client.ontologyOntologyListobjecttypes(),
		client.pipelinePipelineListpipelines(),
		client.adminV2Getcontrolpanel()
	]);

	return {
		datasets: datasets.datasets?.length ?? 0,
		ontologyTypes: ontology.object_types?.length ?? 0,
		pipelines: pipelines.pipelines?.length ?? 0,
		controlPanel
	};
}

File: sdks/python/openfoundry-sdk/examples/phase1_critical_path.py
from __future__ import annotations

from openfoundry_sdk import OpenFoundryClient


def build_client(base_url: str, token: str) -> OpenFoundryClient:
    return OpenFoundryClient(
        base_url=base_url,
        headers={"authorization": f"Bearer {token}"},
    )


def critical_path_calls(client: OpenFoundryClient) -> dict[str, object]:
    return {
        "datasets": client.dataset_dataset_listdatasets,
        "ontology_types": client.ontology_ontology_listobjecttypes,
        "pipelines": client.pipeline_pipeline_listpipelines,
        "control_panel": client.admin_v2_getcontrolpanel,
    }


if __name__ == "__main__":
    client = build_client("https://platform.openfoundry.local", "<token>")
    print(sorted(critical_path_calls(client).keys()))
